### PR TITLE
Replace references to "docker registry" w "container registry" throughout

### PR DESCRIPTION
**Pull request by martyav**
Addressing #229 

I made some simple changes.

I also marked lines with // where I need more help or technical review.

We have several links to the Docker docs, some docker-specific commands, and two Docker-specific subsections, all of which were marked.

### CODE REVIEW

**Comment by martyav** [comment]
Containerd links added. Somebody should check if the content looks relevant enough. I tried to find a good fit.

We still need advice on the Docker-specific commands (add containerd-specific ones as well) and on the Docker-specific subsections (remove, make generic, or keep alongside new container-specific subsections?)

**Comment by Martin-Weiss** [comment]
Just a side note - we do not store "containers" in a registry - we store "images" in a registry. So mayby using "image registry" instead of "container registry" would be the better way?

**Comment by Martin-Weiss** [comment]
Thanks for the research. During workshops with customers it is always a challenge to explain the difference between image and container.. so I would vote for „image registry“.. or maybe remove both and just use „registry“..

**Comment by martyav** [comment]
If this is a pain point when educating customers, maybe we can do "container image registry" on first mention, then use "image registry" for short throughout.

**Comment by martyav** [comment]
In my meeting with Billy, we agreed to leave the Docker specific sections and commands alone. 

I removed the slashes/comments and did a sweep to make sure we're consistently using "container image" or just "image" when referring to registries. I didn't touch the versioned docs. I also did some minor copy edits to a title, a sentence about mirroring registries, and to some link text.

**Comment by martyav** [comment]
@btat this is now ready for review. 

Converting this to a draft after adding you as a reviewer has left the PR in a weird no-man's-land where I can't use the sidebar button to request review now that it's not a draft anymore.

File: docs/reference-guides/cluster-configuration/downstream-cluster-configuration/node-template-configuration/digitalocean.md [hunk 1]
```diff
@@ -10,9 +10,9 @@ The **Droplet Options** provision your cluster's geographical region and specifi
...
 
-The [Docker daemon](https://docs.docker.com/engine/docker-overview/#the-docker-daemon) configuration options include:
+If you use Docker, the [Docker daemon](https://docs.docker.com/engine/docker-overview/#the-docker-daemon) configuration options include:
 
 - **Labels:** For information on labels, refer to the [Docker object label documentation.](https://docs.docker.com/config/labels-custom-metadata/)
 - **Docker Engine Install URL:** Determines what Docker version will be installed on the instance.
 - **Registry mirrors:** Docker Registry mirror to be used by the Docker daemon
-- **Other advanced options:** Refer to the [Docker daemon option reference](https://docs.docker.com/engine/reference/commandline/dockerd/)
\ No newline at end of file
+- **Other advanced options:** Refer to the [Docker daemon option reference](https://docs.docker.com/engine/reference/commandline/dockerd/)
```
Comment (btat) [inline docs/reference-guides/cluster-configuration/downstream-cluster-configuration/node-template-configuration/digitalocean.md:18]:
```suggestion
- **Other advanced options:** Refer to the [Docker daemon option reference](https://docs.docker.com/engine/reference/commandline/dockerd/).
```

**Review by btat — COMMENTED** [review]
LGTM. Just a missing period on one of the bullet points.

**Review by btat — APPROVED** [review]

**Pull request merged**